src/django/pfb_analysis/models.py

```diff
@@ -150,6 +242,16 @@ def logs_url(self):
                           batch_job_name=self.batch_job_name,
                           batch_job_id=self.batch_job_id)
 
+    def cancel(self, reason=None):
+        """ Cancel the analysis job, if its running """
+        if not reason:
+            reason = 'AnalysisJob terminated by user at {}'.format(datetime.utcnow())
+
+        if self.status in self.Status.ACTIVE_STATUSES:
+            client = boto3.client('batch')
+            client.terminate_job(jobId=self.batch_job_id, reason=reason)
```

Comment:
Whoops. Missed that. Will fixup with the if check since that's a specific thing that should be accounted for. I'm not sure we would want to catch the broader botocore error since that could potentially hide other problems. 

It probably makes sense to catch errors in the loop and continue, logging a verbose warning if a particular job fails to cancel. Job ids can still be noted and cancelled manually if necessary via console.